libs/tsconfig.json

```diff
@@ -1,6 +1,6 @@
 {
     "compilerOptions": {
-        "target": "es5",
+        "target": "esnext",
```

Comment:
>esnext [](start = 19, length = 6)

Wondering: why not es2018 here?

Reply:
We have two versions of typescript that we use -- one used locally to build most parts of the webapp (e.g. the app itself, the cli), which is currently 3.7.something, and one that we used in the client itself (or in portions of the code that are trying to match the client), which is 2.7.something (it's here: https://github.com/microsoft/pxt/tree/master/pxtcompiler/ext-typescript) -- this is the one we use for building user programs. The tsconfigs that use esnext are the ones that use the 2.7 compiler; in that one es2018 is 'latest' / 'esnext' because it wasn't completely finalized yet, but the behavior matches properly for our purposes.

I'll file an issue on monday for upgrading the shipped compiler as well -- there are some nice features we're missing out in our editor due to the old version we have, but upgrading is non trivial -- a while back we used the same one we build the app with, but we currently rely on some internal api's in typescript for our static typescript compiler so when we upgraded the webapp to typescript 3 we split off the shipped compiler to keep the old version. Bringing it back up to date + ideally communicating what we need to see if we can use supported api's seems like a worthwhile investment to me (at some point at least..)

(We split it in https://github.com/microsoft/pxt/pull/6046 for reference)

Reply:
That makes sense. I really want the spread operator in editor :)